src/HID.cc

```diff
@@ -45,13 +49,38 @@ HID::HID(const Napi::CallbackInfo &info)
     return;
   }
 
-  if (info.Length() < 1)
+  auto argsLength = info.Length();
+
+  if (argsLength < 1)
   {
     Napi::TypeError::New(env, "HID constructor requires at least one argument").ThrowAsJavaScriptException();
     return;
   }
 
-  if (info.Length() == 1)
+#if defined(__APPLE__)
+  if (argsLength > 1)
+  {
+    // We check if we have an optional param
+    if (info[argsLength - 1].IsObject())
+    {
+      argsLength -= 1;
+      Napi::Object options = info[argsLength].As<Napi::Object>();
+
+      Napi::Value isNonExclusiveMode = options.Get("nonExclusive");
+      if (!isNonExclusiveMode.IsBoolean())
+      {
+        Napi::TypeError::New(env, "nonExclusive flag should be a boolean").ThrowAsJavaScriptException();
+        return;
+      }
+      if (isNonExclusiveMode.As<Napi::Boolean>().Value())
+      {
+        hid_darwin_set_open_exclusive(0);
+      }
+    }
+  }
```

Comment:
```suggestion
  bool isNonExclusiveBool = false
  if (argsLength > 1)
  {
    // We check if we have an optional param
    if (info[argsLength - 1].IsObject())
    {
      argsLength -= 1;
      Napi::Object options = info[argsLength].As<Napi::Object>();

      Napi::Value isNonExclusiveMode = options.Get("nonExclusive");
      if (!isNonExclusiveMode.IsBoolean())
      {
        Napi::TypeError::New(env, "nonExclusive flag should be a boolean").ThrowAsJavaScriptException();
        return;
      }
      isNonExclusiveBool = isNonExclusiveMode.As<Napi::Boolean>().Value();
    }
  }
  
  hid_darwin_set_open_exclusive(isNonExclusiveBool ? 0 : 1);
```

This needs to be able to both enable and disable this flag. And it should do so even when the options object is not defined